build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/Jdk.java

```diff
@@ -152,7 +153,11 @@ public Object getBinJavaPath() {
         return new Object() {
             @Override
             public String toString() {
-                return getHomeRoot() + getPlatformBinPath();
+                try {
+                    return new File(getHomeRoot() + getPlatformBinPath()).getCanonicalPath();
```

Comment:
Ensure we use the canonical path as jdk provisioning is based on canonical path and we would get false negatives otherwise when comparing